bokehjs/src/lib/models/glyphs/glyph.ts

```diff
@@ -213,15 +208,55 @@ export abstract class GlyphView extends View {
 
   protected _project_data(): void {}
 
+  private *_iter_visuals(): Generator<p.VectorSpec<unknown>> {
+    for (const visual of values<visuals.ContextProperties>(this.visuals)) {
+      for (const prop of visual) {
+        if (prop instanceof p.VectorSpec)
+          yield prop
+      }
+    }
+  }
+
+  protected base?: this
+  set_base<T extends this>(base: T): void {
+    if (base != this && base instanceof this.constructor)
+      this.base = base
+  }
+
+  set_visuals(source: ColumnarDataSource, indices: Indices): void {
+    const self = this as any
+
+    for (const prop of this._iter_visuals()) {
+      const {base} = this
+      if (base != null) {
```

Comment:
Can `base` change between iterations?

Reply:
No, in fact base may change only when glyph renderer's glyphs change. However, this condition is a fall-through, so it's easier this way. Anyway, this code will be replaced in future work.